algoliasearch-core/src/test/java/com/algolia/search/JacksonParserTest.java

```diff
@@ -806,4 +809,12 @@ private IndexSettings serializeDeserialize(IndexSettings obj) throws IOException
             serialized,
             Defaults.getObjectMapper().getTypeFactory().constructType(IndexSettings.class));
   }
+
+  @Test
+  void searchSynonymsQueryWithType() throws JsonProcessingException {
+    SynonymQuery query =
+        new SynonymQuery("query").setType(Arrays.asList(ONE_WAY_SYNONYM, ALT_CORRECTION_1));
+    String serialized = Defaults.getObjectMapper().writeValueAsString(query);
+    assertThat(serialized).isEqualTo("{\"query\":\"q\",\"type\":\"oneWaySynonym,altCorrection1\"}");
```

Comment:
Does the test pass? IIRC the default separator for `FlatListSerializer` is a `;`. We could parametrize it.

https://github.com/algolia/algoliasearch-client-java-2/blob/febbaeafd7fbdee4affc9856a4601947a1bd1443/algoliasearch-core/src/main/java/com/algolia/search/models/indexing/FlatListSerializer.java#L19-L24

Reply:
Indeed it does not. My command `mvn -DfailIfNoTests=false -Dtest=com.algolia.search.searchSynonymsQueryWithType test` to run a single test is not correct (hence the test was not run and everything was green).

Reply:
For the record, here's the correct maven command (using `class#method`):
```
mvn -DfailIfNoTests=false -Dtest='JacksonParserTest#searchSynonymsQueryWithType' test
```